### ISSUE TITLE
Unreasonable memory usage

### ISSUE THREAD

**Issue by praisethemoon** [praisethemoon/cgraph#22]
Cannot even run mnist example with 100 images for test. There has to be a leak somewhere in C or the C Lua bridge API.

**Comment by praisethemoon** [comment]
The cause of this issue is `CrossEntropyLoss` function from the C API.

Specifically, the softmax node execution is not freed.

**Comment by praisethemoon** [comment]
As its current state, everything looks good, even from the Lua API.
